File: youtube.py
from selenium import webdriver


if __name__ == '__main__':

    # Chrome WebDriver 로 chrome.exe 실행
    browser = webdriver.Chrome('./chromedriver.exe')

    # 사이트 호출
    browser.implicitly_wait(10)
    browser.get("https://youtube.com")

    # 유튜브 검색창에 검색어 입력
    serachstring = browser.find_element_by_xpath('//*[@id="search"]')
    serachstring.send_keys('아이유')

    # 유튜브 검색버튼 클릭
    serachbutton = browser.find_element_by_xpath('// *[ @ id = "search-icon-legacy"]')
    serachbutton.click()

    # 페이지 읽기
    print(browser.page_source)

    # 드라이브 종료
    browser.quit()

File: golang/youtube.go
package main

import (
	"fmt"
	"time"

	"github.com/tebeka/selenium"
)

const (
	path = `E:\GoApp\src\chromedriver.exe`
	port = 8080
)

func main() {

	opts := []selenium.ServiceOption{}

	server, err := selenium.NewChromeDriverService(path, port, opts...)
	if err != nil {
		fmt.Println(err)
	}
	defer server.Stop()

	c := selenium.Capabilities{"browserName": "chrome"}

	wd, err := selenium.NewRemote(c, fmt.Sprintf("http://localhost:%d/wd/hub", port))
	if err != nil {
		fmt.Println(err)
	}
	defer wd.Quit()

	err = wd.Get("https://youtube.com")
	if err != nil {
		panic(err)
	}

	wd.SetImplicitWaitTimeout(10 * time.Second)

	id, _ := wd.FindElement(selenium.ByCSSSelector, `#search`)
	id.Clear()
	id.SendKeys("아이유")

	searchbtn, _ := wd.FindElement(selenium.ByCSSSelector, `#search-icon-legacy > yt-icon`)
	searchbtn.Click()

	wd.SetImplicitWaitTimeout(10 * time.Second)

	url, _ := wd.CurrentURL()
	title, _ := wd.Title()
	pagesource, _ := wd.PageSource()

	fmt.Println(url)
	fmt.Println(title)
	fmt.Println(pagesource)

	time.Sleep(10 * time.Second)
}
